src/tests/ftest/control/dmg_storage_query.py

```diff
@@ -26,8 +28,46 @@ class DmgStorageQuery(ControlTestBase):
     def setUp(self):
         """Set up for dmg storage query."""
         super().setUp()
-        self.bdev_list = self.server_managers[-1].get_config_value("bdev_list")
-        self.targets = self.server_managers[-1].get_config_value("targets")
+        self.targets = self.server_managers[-1].get_config_value('targets')
+        md_on_ssd = False
+
+        self.log_step('Determining server storage config')
+        bdev_tiers = 0
+        self.bdev_list = []
+        for engine in self.server_managers[-1].manager.job.yaml.engine_params:
+            for index, tier in enumerate(engine.storage.storage_tiers):
+                if tier.storage_class.value == 'ram':
+                    md_on_ssd = True
+                elif tier.storage_class.value == 'nvme':
+                    bdev_tiers += 1
+                    for device in tier.bdev_list.value:
+                        self.bdev_list.append(
+                            {'bdev': device, 'roles': tier.roles.value, 'tier': index})
+        if md_on_ssd:
+            for device in self.bdev_list:
+                if device['roles']:
+                    continue
+                if bdev_tiers == 1 and device['tier'] == 1:
+                    # First bdev storage tier of 1 tier: all roles
+                    device['roles'] = 'wal,data,meta'
+                elif bdev_tiers == 2 and device['tier'] == 1:
+                    # First bdev storage tier of 2 tiers: wal roles
+                    device['roles'] = 'wal'
+                elif bdev_tiers == 2 and device['tier'] == 2:
+                    # Second bdev storage tier of 2 tiers: data & meta roles
+                    device['roles'] = 'data,meta'
+                elif bdev_tiers > 2 and device['tier'] == 1:
+                    # First bdev storage tier of >2 tiers: wal
+                    device['roles'] = 'wal'
+                elif bdev_tiers > 2 and device['tier'] == 2:
+                    # Second bdev storage tier of >2 tiers: meta
+                    device['roles'] = 'meta'
+                else:
+                    # Additional bdev storage tier of >2 tiers: data
+                    device['roles'] = 'data'
```

Comment:
Do we need to be explicit here "just in case"?
```suggestion
                elif bdev_tiers > 2 and device['tier'] == 2:
                    # Second bdev storage tier of >2 tiers: meta
                    device['roles'] = 'meta'
                elif bdev_tiers > 2 and device['tier'] > 2:
                    # Additional bdev storage tier of >2 tiers: data
                    device['roles'] = 'data'
```